File: 프로그래머스/unrated/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.swift
import Foundation

func solution(_ arr:[Int], _ n:Int) -> [Int] {
    var ans = arr
    if arr.count % 2 == 0 {
        for i in 0 ..< arr.count {
            if i % 2 == 1 {
                ans[i] += n 
            }
        }
    } else {
        for i in 0 ..< arr.count {
            if i % 2 == 0 {
                ans[i] += n 
            }
        }
    }
    return ans
}

File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.py
def solution(arr, n):
    answer = arr
    if len(answer) % 2 == 1:
        for i in range(len(answer)):
            if i % 2 == 0:
                answer[i] += n
    else:
        for i in range(len(answer)):
            if i % 2 == 1:
                answer[i] += n
        
    return answer